api/src/auth_routes.ts

```diff
@@ -63,9 +64,35 @@ export function determine_callback_urls(provider_name: string): {
   };
 }
 
+/**
+ * Check that a redirect URL is one that we allow
+ *
+ * @param redirect URL to redirect to
+ * @returns a valid URl to redirect to, default to '/' if
+ *   the one passed in is bad
+ */
+function validateRedirect(redirect: string) {
+  if (redirect.startsWith('http')) {
+    // should match against a whitelist of allowed URLs
+    return redirect;
+  } else if (
+    redirect.startsWith('/') ||
+    redirect.startsWith('org.fedarch.faims3://')
+  ) {
+    return redirect;
+  } else {
+    return '/';
+  }
+}
+
```

Comment:
Could you explain in the doc string what makes a redirect valid? E.g. what is org.fedarch.faims3? Is that something to do with the app?